apis/src/main/java/org/apache/rocketmq/apis/producer/Producer.java

```diff
@@ -0,0 +1,132 @@
+     * @throws AuthorisationException            if no permission to send message.
+     * @throws AuthenticationException           if identification could not be recognized by server.
+     * @throws ProducerClosedAlreadyException    if producer is closed already.
+     * @throws MessageTypeDoesNotMatchException  if message type does not match with the topic.
+     * @throws NetworkTimeoutException           if encountered network timeout to communicate with server.
+     * @throws NetworkConnectionException        if there is a network connection problem.
+     * @throws PersistenceException              if encountered persistence failure from server.
+     * @throws TransactionCheckerNotSetException if {@link TransactionChecker} is not set.
+     */
+    MessageId send(Message message, Transaction transaction) throws ClientException;
+
+    /**
+     * Send a message asynchronously.
+     *
+     * <p>This method returns immediately, the result is included in the {@link CompletableFuture};
+     *
+     * @param message message to send.
+     * @return a future that indicates the result.
+     */
+    CompletableFuture<MessageId> sendAsync(Message message);
+
+    /**
+     * Send batch messages synchronously.
+     *
+     * <p>This method does not return until it gets the definitive result.
+     *
+     * <p>All messages to send should have the same topic.
+     *
+     * @param messages batch messages to send.
+     * @return collection indicates the message id assigned to the appointed message, which keep the same order
+     * messages collection.
+     * @throws TopicDoesNotExistException       if the topic of message does not exist.
+     * @throws AuthorisationException           if no permission to send message.
+     * @throws AuthenticationException          if identification could not be recognized by server.
+     * @throws ProducerClosedAlreadyException   if producer is closed already.
+     * @throws MessageTypeDoesNotMatchException if message type does not match with the topic.
+     * @throws NetworkTimeoutException          if encountered network timeout to communicate with server.
+     * @throws NetworkConnectionException       if there is a network connection problem.
+     * @throws PersistenceException             if encountered persistence failure from server.
+     */
+    Collection<MessageId> send(Collection<Message> messages) throws ClientException;
+
+    /**
+     * Begin transaction, which follows the
+     * <a href="https://en.wikipedia.org/wiki/Two-phase_commit_protocol">two-phase commit protocol</a>.
+     *
+     * @return a transaction entity to execute commit/rollback operation.
+     * @throws TransactionCheckerNotSetException if {@link TransactionChecker} is not set.
+     * @throws ProducerClosedAlreadyException    if producer is closed already.
+     */
+    Transaction beginTransaction() throws ClientException;
```

Comment:
maybe an interface extand producer add some Transaction relate method will be better.

Reply:
Could you please provide some code samples? @WJL3333